twilight-model/src/channel/message/component/mod.rs

```diff
@@ -736,6 +756,9 @@ mod tests {
                 Component::SelectMenu(SelectMenu {
                     channel_types: None,
                     custom_id: "test custom id 2".into(),
+                    default_values: Some(vec![SelectDefaultValue::User(Id::from(
```

Comment:
we might want to test an empty vec, and a vec of multiple elements here

Reply:
Since copying the tokens for each variant would result in many lines of duplicated code, I created a simple "helper function" that dynamically creates the tokens slice used by `serde_test`. However, I had to use `Box::leak` to satisfy `serde_test`'s `'static` requirement.

Does this address your request?

Reply:
thats uncommon but it works in the context of a test, thank you

Reply:
Sorry for the delay. I was rather busy last week.

> we could instead pass the Token as another parameter, duplicating only the id field

@laralove143, I believe e33f30f should implement your approach. Could you please have another look?
As for that commit's test failure, I'm a little confused, as I didn't change twilight-http at all.

Reply:
yes thats what i meant, thank you and no worries about the delay

Reply:
and for the CI, changing models could potentially break http if they're used in http, which they are

but it can also break unexpectedly, maybe because it got rebased on top of other commits one of which is breaking the tests

do the tests pass locally for you? i'll approve once we get the tests to work